### PR TITLE
Add event handlers docs for SDK

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://github.com/metabase/metabase/pull/45153

### Description

Add readme content for `eventHandlers` in `MetabaseProvider` and onLoad / onLoadWithoutCards in Dashboard components

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 1]
```diff
@@ -325,6 +325,8 @@ After the SDK is configured, you can embed your dashboard using the `StaticDashb
 - **withCardTitle**: `boolean` – Whether the dashboard cards should display a title.
 - **withDownloads**: `boolean | null` – Whether to hide the download button.
 - **hiddenParameters**: `string[] | null` – A list of parameters that will not be shown in the set of parameter filters. [More information here](https://www.metabase.com/docs/latest/questions/sharing/public-links#filter-parameters)
+- **onLoad**: `(dashboard: Dashboard | null) => void;` - event handler that triggers after dashboard loads, but without its cards - at this stage dashboard title, tabs and cards grid is rendered, but cards content is not yet loaded.
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/README.md:328]: nit: I feel `onLoad` should be reserved for when everything has been loaded. This is what I'd expect from the API. It's comparable to [load](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) and [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) where `load` is only fired when the whole page including its contents are loaded as opposed to the latter event where only HTML is parsed.

I'd propose `onLoad` to replace `onLoadWithCards` and come up with a new name to replace this _only dashboard has loaded_ event.

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 2]
```diff
@@ -654,6 +658,32 @@ return (
...
 ```
 
+### Adding global event handlers
+
+`MetabaseProvider` also supports `eventHandlers` configuration. This way you can add global handlers to react on events that happen in the SDK context.
+
+Currently, we support:
+
+- `onDashboardLoad?: (dashboard: Dashboard | null) => void;` - triggers when dashboard structure loads (dashboard title, list of tabs, cards grid for the active tab)
+- `onDashboardLoadWithCards?: (dashboard: Dashboard | null) => void;` // triggers when active tab loads with all cards content
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/README.md:668]: Follow the current format.
```suggestion
- `onDashboardLoadWithCards?: (dashboard: Dashboard | null) => void;` - triggers when active tab loads with all cards content
```

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 3]
```diff
@@ -325,6 +325,8 @@ After the SDK is configured, you can embed your dashboard using the `StaticDashb
 - **withCardTitle**: `boolean` – Whether the dashboard cards should display a title.
 - **withDownloads**: `boolean | null` – Whether to hide the download button.
 - **hiddenParameters**: `string[] | null` – A list of parameters that will not be shown in the set of parameter filters. [More information here](https://www.metabase.com/docs/latest/questions/sharing/public-links#filter-parameters)
+- **onLoad**: `(dashboard: Dashboard | null) => void;` - event handler that triggers after dashboard loads with all visible cards and their content.
+- **onLoadWithoutCards**: `(dashboard: Dashboard | null) => void;` - event handler that triggers after dashboard loads, but without its cards - at this stage dashboard title, tabs and cards grid is rendered, but cards content is not yet loaded.
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/README.md:329]: Nice! I love this name change, it's very clear now.

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 4]
```diff
@@ -654,6 +658,33 @@ return (
...
+const eventHandlers = {
+    onDashboardLoad: handleDashboardLoad,
+    onDashboardLoadWithoutCards: handleDashboardLoad,
+}
+
+return (
+  <MetabaseProvider config={config} eventHandlers={eventHandlers}>
+      {children}
+  </MetabaseProvider>
+);
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/README.md:684]: nit: mixed 2-space and 4-space indentation

**Review by heypoom — APPROVED** [review]
Love the name change! 🚀

**Pull request merged**